lib/fastlane/plugin/wpmreleasetoolkit/actions/common/buildkite_upload_pipeline_action.rb

```diff
@@ -0,0 +1,70 @@
+module Fastlane
+  module Actions
+    class BuildkiteUploadPipelineAction < Action
+      def self.run(params)
+        env_file = params[:env_file]
+        pipeline_file = params[:pipeline_file]
+        branch = params[:branch]
+        commit = params[:commit]
+
+        UI.user_error!("Pipeline file not found: #{pipeline_file}") unless File.exist?(pipeline_file)
+
+        UI.message "Uploading pipeline #{pipeline_file} on branch #{branch}, commit #{commit}"
+
+        ENV['BUILDKITE_BRANCH'] = branch
+        ENV['BUILDKITE_COMMIT'] = commit
+
+        if env_file && File.exist?(env_file)
+          UI.message("Sourcing environment file: #{env_file}")
+
+          sh(". #{env_file} && buildkite-agent pipeline upload #{pipeline_file}")
+        else
+          sh('buildkite-agent', 'pipeline', 'upload', pipeline_file)
+        end
+      end
+
+      def self.description
+        # https://buildkite.com/docs/agent/v3/cli-pipeline#uploading-pipelines
+        'Uploads a pipeline to Buildkite, adding all its steps to the current build'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :env_file,
+            description: 'The path to the environment variables file to be sourced before uploading the pipeline',
+            optional: true,
+            type: String
+          ),
```

Comment:
When I saw this `ConfigItem`'s name at first, I thought this was about [the `$BUILDKITE_ENV_FILE` variable exposed by Buildkite itself](https://buildkite.com/docs/pipelines/environment-variables#BUILDKITE_ENV_FILE)… 

Given our conventions I think we might as well call this item `:pipeline_shared_vars`?

Also, given your implementation already takes care of checking if the file exists and ignoring it otherwise, I'd suggest to add `'pipeline-shared-vars'` as a default value for this `ConfigItem`, given that's what our convention uses (and our `local.default.pipeline_upload_steps` on our IaC also hardcodes anyway), so that we don't have to pass it explicitly to all our call sites in our client repo's `Fastfile` every time.

```suggestion
          FastlaneCore::ConfigItem.new(
            key: :pipeline_shared_vars_file,
            description: 'The path to a bash file to be sourced before uploading the pipeline',
            optional: true,
            default_value: 'pipeline-shared-vars',
            type: String
          ),
```

[EDIT] I now see that you force-pushed changes while I was in the middle of this review, and you now renamed it `env_file` instead of `pipeline_env_file`; maybe that new name is ok then. I'd still provide a default value though.

Reply:
> I now see that you force-pushed changes

Sorry about that 😞 -- when I created the PR to test it, I realized the `pipeline_env_file` was a bit convoluted and renamed it back to `env_file`, which was how I started. I think that was the only change after I marked it for Ready For Review?
I assumed it was still "early enough"; but in any case, sorry about that, I should have just gone with a regular commit renaming the param.

Reply:
The problem with defining a `default_value` for `env_file` is that a default would have to assume a location of the file. Maybe something like `../.buildkite/shared-pipeline-vars` would work, but wouldn't that be too specific to be added in `release-toolkit`? 🤔 

For now I've adopted the suggested description on 7349eb36526b2d7e1d0b95beb822d0399ab4a6f4.

Reply:
There's some nuance to this as `pipeline.yml` (and even `.buildkite/pipeline.yml`) are possible [defaults on Buildkite side](https://buildkite.com/docs/agent/v3/cli-pipeline#uploading-pipelines-description).
But in general I agree it's fine to build this kind of conventions (e.g. `.buildkite/shared-pipeline-vars`), also due to the fact this this is a novel / non standard mechanism we built. So I'm fine with it 👍  I'll test it and add it.

By the way, I guess you meant this, right?
```yaml
default_value: `.buildkite/shared-pipeline-vars`, # not './buildkite/pipeline-shared-vars'
```

Reply:
> There's some nuance to this as `pipeline.yml` (and even `.buildkite/pipeline.yml`) are possible [defaults on Buildkite side](https://buildkite.com/docs/agent/v3/cli-pipeline#uploading-pipelines-description).

True. But to contrast this, the use of e.g the `PIPELINE` env var to specify the name of the YAML file to upload is not a default behavior from Buildkite itself, only an artefact of the fact that our steps use that env var in our call to `buildkite-agent pipeline upload ${PIPELINE:-pipeline.yml}` in the main step of all our pipelines; yet we make such assumption in the implementation of `buildkite_trigger_build` action. So all I'm saying is that some of our release-toolkit actions already rely on our own conventions specific to our setup and not just Buildkite's built-in behavior 🙃


> By the way, I guess you meant this, right?
> 
> ```yaml
> 
> default_value: `.buildkite/shared-pipeline-vars`, # not './buildkite/pipeline-shared-vars'
> 
> ```
Yes, of course. I can never remember the exact name of that file from the top of my head without looking it up 😅

Reply:
> True. But to contrast this, the use of e.g the PIPELINE env var to specify the name of the YAML file to upload is not a default behavior from Buildkite itself [...]  all I'm saying is that some of our release-toolkit actions already rely on our own conventions specific to our setup and not just Buildkite's built-in behavior

Yep indeed, agreed.
Thanks for the suggestions! I've added the default `env_file` on 2478ff8437e35e1774ff5c2d2c9a1afe792047cd (with some changes for the tests on 2665c7b67998526236c5149f2621bce01295b8c6).